src/pages/basket-page/components/basket-item.tsx

```diff
@@ -0,0 +1,79 @@
+import { useEffect, useState } from 'react';
+import { generatePath, useNavigate } from 'react-router-dom';
+
+import { EIconType, Icon } from '@/components/icons';
+import { ICON_COLORS } from '@/components/icons/constants';
+import { APP_LINK_URLS, EAppPage } from '@/constants/link-urls';
+import type { SelectedProduct } from '@/interfaces/selected-product';
+
+import type { DeleteSelectedProduct, UpdateSelectedProductCount } from '../typedef';
+
+import styles from './basket-item.module.css';
+
+interface Props {
+    product: SelectedProduct;
+    updateProductCount: UpdateSelectedProductCount;
+    deleteSelectedProduct: DeleteSelectedProduct;
+}
+
+export const BasketItem = ({ product, updateProductCount, deleteSelectedProduct }: Props) => {
+    const { productId, image, title, price } = product;
+
+    const [count, setCount] = useState<number>(product.count);
+    const navigate = useNavigate();
+    useEffect(() => {
+        updateProductCount(productId, count);
+    }, [count, productId, updateProductCount]);
+
+    const incrementCount = () => {
+        setCount((previousCount) => ++previousCount);
+    };
+
+    const decrementCount = () => {
+        setCount((previousCount) => --previousCount);
+    };
+
+    const deleteItem = () => {
+        deleteSelectedProduct(productId);
+    };
+
+    const goToProduct = () => {
+        navigate(generatePath(APP_LINK_URLS[EAppPage.PRODUCT], { id: productId }));
+    };
+    return (
+        <div className={styles['basket__product']} onClick={goToProduct}>
```

Comment:
`div onClick` instead of link -  😫
I suggested adding links to the product's image and title. 
Your approach is also valid, but now the buttons `+` `-` and `🗑️` are not working. On click user is redirected to product's page. Proly you need to stop propagation of these events. 

Also, you can't put one clickable element inside of another by spec. 
Here is a [cool article](https://inclusive-components.design/cards/) about how to create the best layout for the card  
And here is the [solution](https://stackoverflow.com/questions/9882916/are-you-allowed-to-nest-a-link-inside-of-a-link/#9883044)

Reply:
fixed